core/services/ocr2/plugins/ocr2keeper/custom_telemetry.go

```diff
@@ -0,0 +1,179 @@
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+)
+
+const (
+	customTelemChanSize = 100
+)
+
+type AutomationCustomTelemetryService struct {
+	utils.StartStopOnce
+	monitoringEndpoint commontypes.MonitoringEndpoint
+	headBroadcaster    httypes.HeadBroadcaster
+	headCh             chan blockKey
+	unsubscribe        func()
+	chDone             chan struct{}
+	lggr               logger.Logger
+	configDigest       [32]byte
+	latestConfigDigest latestConfigDigestGetter
+}
+
+type latestConfigDigestGetter interface {
+	LatestConfigDetails(opts *bind.CallOpts) (keeper_registry_wrapper2_0.LatestConfigDetails, error)
+}
+
+// NewAutomationCustomTelemetryService creates a telemetry service for new blocks and node version
+func NewAutomationCustomTelemetryService(me commontypes.MonitoringEndpoint, hb httypes.HeadBroadcaster,
+	lggr logger.Logger, chain evm.Chain, rAddr common.Address) (*AutomationCustomTelemetryService, error) {
+	registry, rErr := keeper_registry_wrapper2_0.NewKeeperRegistry(rAddr, chain.Client())
+	if rErr != nil {
+		return nil, errors.Wrap(rErr, "error creating new Registry Wrapper for customTelemService")
+	}
+	return &AutomationCustomTelemetryService{
+		monitoringEndpoint: me,
+		headBroadcaster:    hb,
+		headCh:             make(chan blockKey, customTelemChanSize),
+		chDone:             make(chan struct{}),
+		lggr:               lggr.Named("Automation Custom Telem"),
+		latestConfigDigest: registry,
+	}, nil
+}
+
+// Start starts Custom Telemetry Service, sends 1 NodeVersion message to endpoint at start and sends new BlockNumber messages
+func (e *AutomationCustomTelemetryService) Start(ctx context.Context) error {
+	return e.StartOnce("AutomationCustomTelemetryService", func() error {
+		e.lggr.Infof("Starting: Custom Telemetry Service")
+		callOpt := &bind.CallOpts{Context: ctx}
+		configDetails, cdErr0 := e.latestConfigDigest.LatestConfigDetails(callOpt)
+		if cdErr0 != nil {
+			e.lggr.Errorf("Error occurred while getting newestConfigDetails for initialization %v", cdErr0)
+		} else {
+			e.configDigest = configDetails.ConfigDigest
+			e.sendNodeVersionMsg()
+		}
+		go func() {
+			ticker := time.NewTicker(1 * time.Minute)
+			defer ticker.Stop()
+			for {
+				select {
+				case <-ticker.C:
+					callOpt := &bind.CallOpts{Context: ctx}
+					newConfigDetails, cdErr := e.latestConfigDigest.LatestConfigDetails(callOpt)
+					if cdErr != nil {
+						e.lggr.Errorf("Error occurred while getting newestConfigDetails  %v", cdErr)
+						continue
+					}
+					newConfigDigest := newConfigDetails.ConfigDigest
+					if newConfigDigest != e.configDigest {
+						e.configDigest = newConfigDigest
+						e.sendNodeVersionMsg()
+					}
+				case <-ctx.Done():
+					return
+				}
+			}
+		}()
+		_, e.unsubscribe = e.headBroadcaster.Subscribe(&headWrapper{e.headCh})
+		go func() {
+			e.lggr.Infof("Started: Sending BlockNumber Messages")
+			for {
+				select {
+				case blockInfo := <-e.headCh:
```

Comment:
do we want to send it for all block numbers? Can it overload telemetry on fast chains?